client/lib/create-selector/test/index.js

```diff
@@ -277,6 +301,62 @@ describe( 'index', () => {
 			.true;
 	} );
 
+	describe( 'when no custom cache key generating function is given', () => {
+		test( 'should to create a default key by joining args', () => {
+			const memoizedSelector = createSelector( selector );
+
+			memoizedSelector( {}, 'a', 'b', 'c', 1, 2, 3 );
+
+			expect( memoizedSelector.memoizedSelector.cache.has( 'a,b,c,1,2,3' ) ).to.be.true;
+		} );
+	} );
+
+	describe( 'when "cachePerKey" argument is "true"', () => {
+		test( 'should clear cache when dependants do not match, but only per key', () => {
+			const memoizedSelector = createSelector(
+				( state, x, y, z ) => state[ x ][ y ][ z ],
+				state => state.a.b,
+				null,
+				true
+			);
+			const state = {
+				a: {
+					b: {
+						c: 'cc',
+						d: 'dd',
+					},
+				},
+			};
+
+			memoizedSelector( state, 'a', 'b', 'c' );
+
+			expect( memoizedSelector.memoizedSelector.cache.get( 'a,b,c' ) ).to.eql( 'cc' );
```

Comment:
Minor: `eql` is a deep equality check, which is not necessary when comparing strings, instead comparing strictly via `equal`.

Aside: I thought we were using Jest now? Equivalents would be `.toEqual` (deep) and `.toBe` (shallow)